src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -367,11 +368,24 @@ public override SyntaxNode AsPublicInterfaceImplementation(SyntaxNode declaratio
             });
         }
 
+        public interface Foo
+        {
+            void Moo<T>() where T : IEnumerable<T>;
+        }
+
+        public class Bar : Foo
+        {
+            public void Moo<T>() where T : IEnumerable<T>
+            {
+            }
+        }
+
         public override SyntaxNode AsPrivateInterfaceImplementation(SyntaxNode declaration, SyntaxNode interfaceTypeName, string interfaceMemberName)
```

Comment:
That's true for C#, but not for VB.  VB always implements interface methods explicitly.